src/main/java/com/mongodb/hibernate/internal/translate/mongoast/filter/AstComparisonFilterOperation.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.translate.mongoast.filter;
+
+import com.mongodb.hibernate.internal.translate.mongoast.AstValue;
+import org.bson.BsonWriter;
+
+public record AstComparisonFilterOperation(AstComparisonFilterOperator operator, AstValue value)
```

Comment:
In match syntax, the value of a comparison filter has to be a constant, e.g. `1` or `"1"`.  In the C# implementation this is enforced by requiring `value` to be of type `BsonValue`.  We can't quite do the same here because we also have to support `AstPlaceholder`, but perhaps we can introduce a super-interface that both `AstPlaceholder` and `AstLiteralValue` extend, and use that as the generic type for `AstVisitorValueDescriptor#FIELD_VALUE`.

Reply:
Just a naive questoin. Why did we introduce `AstValue` in the first place? If we used `BsonValue` directly as in C#, it seems to simplify a lot. Maybe `AstPlaceHolder` (now renamed to `AstParameterMarker`) is the reason?

Reply:
but if we treat `AstParameterMarker` as a synonym to `new BsonUndefined()`, seems it works. From my perspective, there is no need to make it inherit `AstNode` (or any other `AstValuel`), for the criterion is whether we want to render it specially. `BsonValue` rendering seems straightforward without need of special handling?

Reply:
> In match syntax, the value of a comparison filter has to be a constant, e.g. `1` or `"1"`. In the C# implementation this is enforced by requiring `value` to be of type `BsonValue`. We can't quite do the same here because we also have to support `AstPlaceholder`, but perhaps we can introduce a super-interface that both `AstPlaceholder` and `AstLiteralValue` extend, and use that as the generic type for `AstVisitorValueDescriptor#FIELD_VALUE`.

Technically speaking, I think `AstPlaceHolder` is kind of `AstLiteralValue` from translating perspective (its special meaning only manifests itself in JDBC `PreparedStatement` layer). If we do this, we could simply by using the generic `AstLiteralValue` for `AstVisitorValueDescriptor#FIELD_VALUE` elegantly.

Reply:
Oh, I'm so sorry.  This already works exactly as I wanted.  I must have mis-read `AstValue` as `AstNode`.  `AstValue` is exactly the supertype I was asking for.